File: 29Items-Projects/7Algorithms/1-Sorting-and-Searching/src/python/algorithms/sorting/selection_sort.py
def selection_sort(values: list[int]) -> list[int]:
    result = values[:]
    for i in range(len(result)):
        min_index = i
        for j in range(i + 1, len(result)):
            if result[j] < result[min_index]:
                min_index = j
        result[i], result[min_index] = result[min_index], result[i]
    return result


if __name__ == "__main__":
    sample = [42, 7, 19, 3, 7, 99, 1]
    print("SelectionSort")
    print(f"input : {sample}")
    print(f"output: {selection_sort(sample)}")

File: 29Items-Projects/7Algorithms/1-Sorting-and-Searching/src/java/com/sortingsearching/sorting/SelectionSort.java
package com.sortingsearching.sorting;

import java.util.Arrays;

public final class SelectionSort {
    private SelectionSort() {
    }

    public static int[] sort(int[] input) {
        int[] values = Arrays.copyOf(input, input.length);
        for (int i = 0; i < values.length; i++) {
            int minIndex = i;
            for (int j = i + 1; j < values.length; j++) {
                if (values[j] < values[minIndex]) {
                    minIndex = j;
                }
            }
            int temp = values[i];
            values[i] = values[minIndex];
            values[minIndex] = temp;
        }
        return values;
    }

    public static void main(String[] args) {
        int[] sample = {42, 7, 19, 3, 7, 99, 1};
        System.out.println("SelectionSort");
        System.out.println("input : " + Arrays.toString(sample));
        System.out.println("output: " + Arrays.toString(sort(sample)));
    }
}
